### PR TITLE
fix(devtools-view): Theme selection dropdown in settings does not keep currently selected value

### PR DESCRIPTION
**Pull request by jikim-msft**
#### Description

https://dev.azure.com/fluidframework/internal/_workitems/edit/4671/

Devtools supports `Light`, `Dark`, and `High Contrast` themes for the users. Currently, the dropdown menu does not retain the selected value if navigated to a different view. This PR fixes this issue and makes users to see their currently selected theme even after going through different panels.

### CODE REVIEW

**Comment by alexvy86** [comment]
@jikim-msft let's write an actual description here. The link to the ADO item is convenient but PRs should ideally be self-contained and understandable by anyone who can see them, including the open source community.

File: packages/tools/devtools/devtools-view/src/components/SettingsView.tsx
Comment (alexvy86) [inline packages/tools/devtools/devtools-view/src/components/SettingsView.tsx:120]: Hmm are we sure this is the correct way? Does `Dropdown` have a prop for "currently selected" or something along those lines?

Comment (jikim-msft) [reply packages/tools/devtools/devtools-view/src/components/SettingsView.tsx:120]: I initially tried using `useState` to store the value that has been selected by the `<Dropdown>` option. So something like:
- Create `const [selectedTheme, setSelectedTheme] = useState("Theme")`
- Trigger `onClick()` event to set the `selectedTheme` correspondingly
- Render `selectedTheme` in our `placeholder`

However, even with this change, the problem persisted and I believe this is due to the way how `View` component (from the `DevtoolsView`) selects its "to-be" rendered component through switch case and initializes the state back to the default selected placeholder. 

Therefore, I instead chose to use `React.context()` which is propagated gloabally from parent to child component to access the currently selected theme.

Comment (alexvy86) [reply packages/tools/devtools/devtools-view/src/components/SettingsView.tsx:120]: But that's separate from my point. Using the `placeholder` property of the `Dropdown` is what I find strange. There must be a way to bind the current value to the actual selected value in the dropdown, not just "fake it" by writing it to the placeholder text which is only supposed to appear when there's no selection. In our app I think "no selection" doesn't make sense. There's always a selected theme; we decided that Dark is the default, and that should mean that when the user comes to the Settings view, that theme should already be selected.

Comment (jikim-msft) [reply packages/tools/devtools/devtools-view/src/components/SettingsView.tsx]: Change using to `value` as it seems more logical than using `placeholder`. But I am still skeptical (and I think this goes along with your point) of using `themeReadableOption[themeInfo.name as ThemeName]` as it feels like our `SettingsView` now has two different sources of data.

Comment (alexvy86) [reply packages/tools/devtools/devtools-view/src/components/SettingsView.tsx]: I agree. The problem there is that when using `setTheme()` in the switch statement we're setting values different than the ones from the `ThemeOption` enum, so then we have to "translate back" from those to the ones in `ThemeOption`, which you did with `themeReadableOption`. If we update the `setTheme()` lines to something like 

```typescript
setTheme({
	name: ThemeOption.Light, // instead of the string "light"
	theme: webLightTheme,
});
```

then I think the code will be more straightforward. I'm only slightly concerned about the fact that the value of one of the options in `ThemeOption` is "High Contrast" with a space in the middle but I think it should be fine... If you do this change let's just make sure that theme selection still works for all three themes.

File: packages/tools/devtools/devtools-view/src/components/SettingsView.tsx
Comment (alexvy86) [inline packages/tools/devtools/devtools-view/src/components/SettingsView.tsx:109]: This is probably because in ThemeHelper.ts we do [this](https://github.com/microsoft/FluidFramework/blob/d8db010073e26f696a3eddfd5fcb71c3aa9a4499/packages/tools/devtools/devtools-view/src/ThemeHelper.ts#L22-L42), right? Let's fix that in this PR too; that code should use the enum as well. With that update this should end up being just `value={themeInfo.name}`, I think.

**Review by alexvy86 — APPROVED** [review]
One comment below that I think we can address before merging to get rid of a small "hack", but otherwise LGTM!

File: packages/tools/devtools/devtools-view/src/components/SettingsView.tsx
Comment (jikim-msft) [reply packages/tools/devtools/devtools-view/src/components/SettingsView.tsx:109]: Yes. It's because the `getFluentUIThemeToUse` has `dark` (with lower-cased "d") as its `name`, and our `SettingsView` component takes "dark" as its initial value. 

I wasn't so sure of the scope of this PR and whether I should be fixing some of the codes from `ThemeHelper` 😃 .

Comment (alexvy86) [reply packages/tools/devtools/devtools-view/src/components/SettingsView.tsx:109]: Maybe ideally we would make those changes (use the enum everywhere _today_) in a separate PR, but it's close enough to this one that I think it's ok to bundle together. I expect those fixes to be relatively minor and they contribute to make the actual change from this PR as clean as we can make it. If making those adjustments starts to get complicated, then we can split it into its own task; but if it'll be an hour or two, let's include it here.

**Pull request merged**
